pymongo/collection.py

```diff
@@ -491,7 +492,7 @@ def insert_many(self, documents, ordered=True):
 
         .. versionadded:: 3.0
         """
-        if not isinstance(documents, list) or not documents:
+        if not isinstance(documents, (list, GeneratorType)) or not documents:
```

Comment:
I think it might make more sense to get rid of list and instead do isinstance(documents, Iterable). Since we validate the documents we really only need to be sure we can iterate "documents".